packages/field-plugin/src/createFieldPlugin/createPluginActions/createPluginActions.ts

```diff
@@ -31,37 +33,37 @@ export const defaultState: FieldPluginData = {
   token: undefined,
   uid: '-preview',
   spaceId: undefined,
+})
+
+export type CreatePluginActionsOptions<Content> = {
+  uid: string
+  postToContainer: (message: unknown) => void
+  onUpdateState: (state: FieldPluginData<Content>) => void
+  parseContent: (content: unknown) => Content
 }
 
-export type CreatePluginActions = (
-  uid: string,
-  postToContainer: (message: unknown) => void,
-  onUpdateState: (state: FieldPluginData) => void,
-) => {
+export const createPluginActions = <Content>(
+  options: CreatePluginActionsOptions<Content>,
+): {
   // These functions are to be called by the field plugin when the user performs actions in the UI
-  actions: FieldPluginActions
+  actions: FieldPluginActions<Content>
   // These functions are called when the plugin receives messages from the container
   messageCallbacks: PluginMessageCallbacks
-}
-
-export const createPluginActions: CreatePluginActions = (
-  uid,
-  postToContainer,
-  onUpdateState,
-) => {
+} => {
+  const { uid, postToContainer, onUpdateState, parseContent } = options
   // Tracks the full state of the plugin.
   //  Because the container doesn't send the full state in its messages, we need to track it ourselves.
   //  isModal and height is not included in the messages to the children and must thus be tracked here.
   //  In future improved versions of the plugin API, this should not be needed.
-  let state: FieldPluginData = defaultState
+  let state: FieldPluginData<Content> = defaultState(parseContent(undefined))
 
   let assetSelectedCallbackRef: undefined | ((filename: Asset) => void) =
     undefined
 
   const onStateChange: OnStateChangeMessage = (data) => {
     state = {
       ...state,
-      ...partialPluginStateFromStateChangeMessage(data),
+      ...partialPluginStateFromStateChangeMessage(data, parseContent),
```

Comment:
The content gets parsed only the first time when the content is received from the visual editor. After that, we only allow the user to update the content with the right type, so we don't need to validate the structure.